### ISSUE TITLE
`:FindFirstChild()` and similar functions shouldn't override the type when the companion studio plugin is on

### ISSUE THREAD

**Issue by Aspecky** [JohnnyMorganz/luau-lsp#543]
The `.` operator gives proper autocompletes about the children of an instance when the companion studio plugin is on, but when getting the children using `:FindFirstChild` or similar, the type gets overridden which makes you lose autocompletes:

```lua
local template = workspace.TemplateR15
local head = template.Head -- gets proper autocompletes and is typed correctly

local template = workspace:FindFirstChild("TemplateR15")
local head = template.Head -- type is *error-type*
```

Roblox studio also does respect those functions along with their arguments whereas luau-lsp doesnt, e.g:
```lua
--[[
    Animator is not a direct child of TemplateR15, it is a descendant of it
    |_ TemplateR15
        |_ Humanoid
        |   |_ Animator
        |_ ...
]]
local template = workspace.TemplateR15
local animator = template:FindFirstChild("Animator") -- type is Instance?

local template = workspace.TemplateR15
local animator = template:FindFirstChild("Animator", true) -- type is Animator
```
This logic applies to all the functions that traverse the tree of an instance.

Current workaround is to cast the type using `typeof(Instance.Child)` but that can get verbose very quick.

**Comment by JohnnyMorganz** [comment]
This is irrelevant to the companion studio plugin.

We don't provide special "FindFirstChild" types. Maybe we should though.

**Comment by JohnnyMorganz** [comment]
Sorry for the delay in this. Your second comment will be under the #685 issue.
I will also fix the first part for this. I will make a separate issue (#689) about supporting recursive find first childs.

**Comment by JohnnyMorganz** [comment]
Fixed by f657718dcb1d709e82cb7202e242b34530cd7d6c
